File: src/main/java/icu/samnyan/aqua/api/model/resp/sega/chuni/v2/external/Chu3DataExport.java
package icu.samnyan.aqua.api.model.resp.sega.chuni.v2.external;

import icu.samnyan.aqua.sega.chusan.model.userdata.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.List;

/**
 * This class is use for exporting chusan profile
 * @author samnyan (publicamusement@protonmail.com)
 */
@Data
@AllArgsConstructor
@NoArgsConstructor
public class Chu3DataExport {
    public String gameId = "SDHD";
    public UserData userData;
    public UserGameOption userGameOption;
    public List<UserActivity> userActivityList;
    public List<UserCharacter> userCharacterList;
    public List<UserCharge> userChargeList;
    public List<UserCourse> userCourseList;
    public List<UserDuel> userDuelList;
    public List<UserItem> userItemList;
    public List<UserMapArea> userMapList;
    public List<UserMusicDetail> userMusicDetailList;
    public List<UserPlaylog> userPlaylogList;
    public List<UserLoginBonus> userLoginBonusList;
}

File: src/main/java/icu/samnyan/aqua/api/model/resp/sega/chuni/v2/external/Chu3DataExport.kt
package icu.samnyan.aqua.api.model.resp.sega.chuni.v2.external

import icu.samnyan.aqua.net.games.IExportClass
import icu.samnyan.aqua.sega.chusan.model.userdata.*

data class Chu3DataExport(
    override var gameId: String = "SDHD",
    override var userData: Chu3UserData,
    var userGameOption: UserGameOption,
    var userActivityList: List<UserActivity>,
    var userCharacterList: List<UserCharacter>,
    var userChargeList: List<UserCharge>,
    var userCourseList: List<UserCourse>,
    var userDuelList: List<UserDuel>,
    var userItemList: List<UserItem>,
    var userMapList: List<UserMap>,
    var userMusicDetailList: List<UserMusicDetail>,
    var userPlaylogList: List<UserPlaylog>,
): IExportClass<Chu3UserData> {
    constructor() : this("SDHD",
        Chu3UserData(), UserGameOption(), ArrayList(), ArrayList(), ArrayList(), ArrayList(), ArrayList(), ArrayList(), ArrayList(), ArrayList(), ArrayList())
}

